File: Sources/CountriesFlagsObjC/source/Countries/UIImage+Thailand.m
//
//  UIImage+Thailand.m
//  CountriesFlags
//

#import "Countries/UIImage+Thailand.h"

@implementation UIImage (Thailand)

+(UIImage * _Nullable) flagThailandWithSize:(CGSize)size {
    UIColor* colorRed = [UIColor colorWithRed:165.0f/255.0f
                                        green:25.0f/255.0f
                                         blue:49.0f/255.0f
                                        alpha:1.0f];
    UIColor* colorWhite = [UIColor colorWithRed:244.0f/255.0f
                                          green:245.0f/255.0f
                                           blue:248.0f/255.0f
                                          alpha:1.0f];
    UIColor* colorBlue = [UIColor colorWithRed:45.0f/255.0f
                                         green:42.0f/255.0f
                                          blue:74.0f/255.0f
                                         alpha:1.0f];
    
    UIGraphicsBeginImageContextWithOptions(size, NO, 0.0f);
    
    [colorRed setFill];
    UIRectFill(CGRectMake(0.0f, 0.0f, size.width, size.height));
    
    [colorWhite setFill];
    UIRectFill(CGRectMake(0.0f, size.height / 6.0f, size.width, size.height / 6.0f * 4.0f));
    
    [colorBlue setFill];
    UIRectFill(CGRectMake(0.0f, size.height / 6.0f * 2.0f, size.width, size.height / 6.0f * 2.0f));
    
    UIImage* image = UIGraphicsGetImageFromCurrentImageContext();
    UIGraphicsEndImageContext();
    
    return image;
}

@end

File: Sources/CountriesFlags/Countries/UIImage+Thailand.swift
//
//  UIImage+Thailand.swift
//  CountriesFlags
//

#if canImport(UIKit)

import UIKit

extension UIImage {
    /**
     [wikipedia]: https://en.wikipedia.org/wiki/Flag_of_Thailand "wikipedia"
     
     Generates the flag of Thailand.
     For more information see [wikipedia].
     
     - Parameters:
        - size: image size.
     
     - Returns: Returns the flag of Thailand.
     */
    internal static func flagThailand(with size: CGSize) -> UIImage {
        let colorRed = UIColor(red: 165.0 / 255.0,
                               green: 25.0 / 255.0,
                               blue: 49.0 / 255.0,
                               alpha: 1.0)
        let colorWhite = UIColor(red: 244.0 / 255.0,
                                 green: 245.0 / 255.0,
                                 blue: 248.0 / 255.0,
                                 alpha: 1.0)
        let colorBlue = UIColor(red: 45.0 / 255.0,
                                green: 42.0 / 255.0,
                                blue: 74.0 / 255.0,
                                alpha: 1.0)

        UIGraphicsBeginImageContextWithOptions(size, true, 0.0)

        colorRed.setFill()
        UIRectFill(CGRect(origin: .zero, size: size))

        colorWhite.setFill()
        UIRectFill(CGRect(origin: CGPoint(x: 0, y: size.height / 6.0),
                          size: CGSize(width: size.width, height: size.height / 6.0 * 4.0)))

        colorBlue.setFill()
        UIRectFill(CGRect(origin: CGPoint(x: 0, y: size.height / 6.0 * 2.0),
                          size: CGSize(width: size.width, height: size.height / 6.0 * 2.0)))

        let image = UIGraphicsGetImageFromCurrentImageContext()
        UIGraphicsEndImageContext()

        return image ?? UIImage()
    }
}

#endif
